sdks/python/apache_beam/transforms/ptransform.py

```diff
@@ -849,8 +854,9 @@ def element_type(side_input):
         if not typehints.is_consistent_with(bindings.get(arg, typehints.Any),
                                             hint):
           raise TypeCheckError(
-              'Type hint violation for \'%s\': requires %s but got %s for %s' %
-              (self.label, hint, bindings[arg], arg))
+              'Type hint violation for \'%s\': requires %s but got %s for %s\n'
+              'Full type hint:\n%s' %
+              (self.label, hint, bindings[arg], arg, type_hints.debug_str()))
```

Comment:
nit: with this many args, either using `.format` with named params or `%(name)d` with an arg dict would be more readable

Reply:
done